code/lib/preview-api/src/modules/preview-web/render/StoryRender.ts

```diff
@@ -179,62 +184,65 @@ export class StoryRender<TRenderer extends Renderer> implements Render<TRenderer
+      kind: title,
+      id,
+      name,
+      story: name,
+      tags,
+      ...this.callbacks,
+      showError: (error) => {
+        this.phase = 'errored';
+        return this.callbacks.showError(error);
+      },
+      showException: (error) => {
+        this.phase = 'errored';
+        return this.callbacks.showException(error);
+      },
+      forceRemount: forceRemount || this.notYetRendered,
+      storyContext: context,
+      storyFn: () => unboundStoryFn(context),
+      unboundStoryFn,
+    };
+
     try {
-      let loadedContext: Awaited<ReturnType<typeof applyLoaders>>;
       await this.runPhase(abortSignal, 'loading', async () => {
-        loadedContext = await applyLoaders({
-          ...this.storyContext(),
-          viewMode: this.viewMode,
-          // TODO add this to CSF
-          canvasElement,
-        } as unknown as StoryContextForLoaders<TRenderer>);
+        context.loaded = await applyLoaders(context);
       });
-      if (abortSignal.aborted) return;
 
-      const renderStoryContext: StoryContext<TRenderer> = {
-        ...loadedContext!,
-        // By this stage, it is possible that new args/globals have been received for this story
-        // and we need to ensure we render it with the new values
-        ...this.storyContext(),
-        abortSignal,
-        // We should consider parameterizing the story types with TRenderer['canvasElement'] in the future
-        canvasElement: canvasElement as any,
-      };
+      if (abortSignal.aborted) return;
 
       await this.runPhase(abortSignal, 'beforeEach', async () => {
-        const cleanupCallbacks = await applyBeforeEach(renderStoryContext);
+        const cleanupCallbacks = await applyBeforeEach(context);
         this.store.addCleanupCallbacks(story, cleanupCallbacks);
       });
 
       if (abortSignal.aborted) return;
 
-      const renderContext: RenderContext<TRenderer> = {
-        componentId,
-        title,
-        kind: title,
-        id,
-        name,
-        story: name,
-        tags,
-        ...this.callbacks,
-        showError: (error) => {
-          this.phase = 'errored';
-          return this.callbacks.showError(error);
-        },
-        showException: (error) => {
-          this.phase = 'errored';
-          return this.callbacks.showException(error);
-        },
-        forceRemount: forceRemount || this.notYetRendered,
-        storyContext: renderStoryContext,
-        storyFn: () => unboundStoryFn(renderStoryContext),
-        unboundStoryFn,
-      };
-
-      await this.runPhase(abortSignal, 'rendering', async () => {
-        const teardown = await this.renderToScreen(renderContext, canvasElement);
-        this.teardownRender = teardown || (() => {});
-      });
+      if (!playFunction || !getUsedProps(playFunction).includes('mount')) {
+        await context.mount();
+        context.mount = async () => {
+          throw new Error('Destructure mount if you want to use it in the play function.');
```

Comment:
Can you make the error a bit more descriptive? https://github.com/storybookjs/storybook/blob/next/code/lib/core-events/src/errors/README.md#how-to-write-a-proper-error-message

Reply:
will do